tests/sentry/rules/processing/test_delayed_processing.py

```diff
@@ -94,8 +96,12 @@ def create_event(
         if tags:
             data["tags"] = tags
 
-        return self.store_event(
-            data=data, project_id=project_id, assert_no_errors=False, event_type=EventType.ERROR
+        # Cast the event to GroupEvent to avoid type errors
```

Comment:
😬 
what type does `store_event` typically produce?

are there any concerns with casting this?

Reply:
`store_event` returns an `Event` type. i went down the rabbit hole of correctly typing that function, but it created hundreds / thousands of type issues 😭 then... i created this helper to return a group event (which broke lots of things) instead of just casting it, so then i updated this function to return a tuple of the event / group event -- which worked, but it was kinda a nightmare to look at in code, and i updated everywhere outside of this function to be a GroupEvent (but that made my actual changes really hard to grok). 😵 

finally i landed on just wrapping this with the cast, so it will be explicit that it's _not_ actually a group event (i spent a long time trying to figure out what was wrong with my test instead because the return type was just set, and it was implicitly casting it.. so at least now it's explicit.

Reply:
It returns the `Event` type. The reason why we're casting is if you don't specify the `GroupEvent`, you have to assert event.group before you can fetch the value b/c it's typed as `Group | None`. We check that everywhere so there'd be a million asserts all over the place